source/blog/2016-12-12-part2-install-origin-on-f25-atomic-host.html.md

```diff
@@ -0,0 +1,353 @@
+	* A source build using source code from https://github.com/openshift/ruby-ex.git will be created
+	  * The resulting image will be pushed to image stream "ruby-ex:latest"
+	  * Every time "ruby-22-centos7:latest" changes a new build will be triggered
+	* This image will be deployed in deployment config "ruby-ex"
+	* Port 8080/tcp will be load balanced by service "ruby-ex"
+	  * Other containers can access this service through the hostname "ruby-ex"
+
+--> Creating resources with label app=ruby-ex ...
+	imagestream "ruby-22-centos7" created
+	imagestream "ruby-ex" created
+	buildconfig "ruby-ex" created
+	deploymentconfig "ruby-ex" created
+	service "ruby-ex" created
+--> Success
+	Build scheduled, use 'oc logs -f bc/ruby-ex' to track its progress.
+	Run 'oc status' to view your app.
+```
+
+Let's take a moment to digest that. A new [image
+stream](https://docs.openshift.org/1.2/architecture/core_concepts/builds_and_image_streams.html#image-streams)
+was created to track the upstream `ruby-22-centos7:latest` image. A
+`ruby-ex`
+[buildconfig](https://docs.openshift.org/1.2/dev_guide/builds.html#defining-a-buildconfig)
+was created that will perform an
+[S2I](https://docs.openshift.org/1.2/architecture/core_concepts/builds_and_image_streams.html#source-build)
+build that will bake the source code into the image from the
+`ruby-22-centos7` image stream. The resulting image will be the source
+for another image stream known as `ruby-ex`. A
+[deploymentconfig](https://docs.openshift.org/1.2/architecture/core_concepts/deployments.html#deployments-and-deployment-configurations)
+was created to deploy the application into pods once the build is done.
+Finally, a `ruby-ex` service was created so the application can be load
+balanced and discoverable.
+
+After a short time, we check the status of the application:
+
+```
+$ oc status 
+In project myproject on server https://54.175.0.44:8443
+
+svc/ruby-ex - 172.30.213.94:8080
+  dc/ruby-ex deploys istag/ruby-ex:latest <-
+	bc/ruby-ex source builds https://github.com/openshift/ruby-ex.git on istag/ruby-22-centos7:latest 
+	  build #1 running for 26 seconds
+	deployment #1 waiting on image or update
+
+1 warning identified, use 'oc status -v' to see details.
+```
+
+We can see here that there is a svc (service) that is associated with a
+dc (deploymentconfig) that is associated with a bc (buildconfig) that
+has a build that has been `running for 26 seconds`. The deployment is
+waiting for the build to finish before attempting to run.
+
+After some more time:
+
+```
+$ oc status 
+In project myproject on server https://54.175.0.44:8443
+
+svc/ruby-ex - 172.30.213.94:8080
+  dc/ruby-ex deploys istag/ruby-ex:latest <-
+	bc/ruby-ex source builds https://github.com/openshift/ruby-ex.git on istag/ruby-22-centos7:latest 
+	deployment #1 running for 6 seconds
+
+1 warning identified, use 'oc status -v' to see details.
+```
+
+The build is now done and the deployment is running.
+
+Tick Tock. Tick Tock:
```

Comment:
Yah, keep in mind that people are going to be reading this for whom English is not a first language.